server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -59,13 +59,22 @@
 import static java.util.Map.entry;
 import static org.elasticsearch.core.Strings.format;
 
+/**
+ * Manages all the Java thread pools we create. {@link Names} contains a list of the thread pools, but plugins can dynamically add more
+ * thread pools to instantiate.
+ */
 public class ThreadPool implements ReportingService<ThreadPoolInfo>, Scheduler {
 
     private static final Logger logger = LogManager.getLogger(ThreadPool.class);
 
     public static class Names {
+        // Often used for quick operations not worth the costs of switching to another thread. Also used to avoid the cost of re-queuing
+        // work while leveraging the interface for ease of coding.
+        // This should not be used on Netty transport threads if the task may not always complete quickly.
         public static final String SAME = "same";
+        // All the tasks that do not fit into the other categories. Try to avoid this if possible.
         public static final String GENERIC = "generic";
+        // Important management tasks that keep the cluster from falling apart. We do not want these operations stalled.
```

Comment:
Could you clarify what you mean by "stalled"? For instance we do some IO on this threadpool, e.g. writing the cluster state to disk (or blocking while it's written to the blob store).

In practice the reason we use this one-thread threadpool is to avoid unnecessary contention on `Coordinator#mutex`: most (but not quite all) of the `synchronized` blocks on this mutex run on this threadpool.